sub/osd_libass.c

```diff
@@ -184,31 +184,38 @@ static void clear_ass(struct ass_state *ass)
 
 void osd_get_function_sym(char *buffer, size_t buffer_size, int osd_function)
 {
-    // 0xFF is never valid UTF-8, so we can use it to escape OSD symbols.
-    // (Same trick as OSD_ASS_0/OSD_ASS_1.)
-    snprintf(buffer, buffer_size, "\xFF%c", osd_function);
+    snprintf(buffer, buffer_size, OSD_CODEPOINTS_ESCAPE "%c", osd_function);
 }
 
 void osd_mangle_ass(bstr *dst, const char *in, bool replace_newlines)
 {
     const char *start = in;
     bool escape_ass = true;
+    size_t len = mp_str_len(OSD_CODEPOINTS_ESCAPE);
     while (*in) {
         // As used by osd_get_function_sym().
-        if (in[0] == '\xFF' && in[1]) {
+        if (!strncmp(in, OSD_CODEPOINTS_ESCAPE, len) && in[len]) {
```

Comment:
nothing resets `len` so this might use a wrong one on later iterations

Reply:
Fixed, I moved it to far, strange.